src/json-patch-duplex.ts

```diff
@@ -524,6 +525,65 @@ module jsonpatch {
     return true;
   }
 
+  
+  /**
+     * Apply a single json-patch on an object tree
+     * Returns the result object.
+     */
+  export function applyPatch<T>(tree: T, patch: Patch<any>): T;
+  export function applyPatch<T>(tree: T, patch: Patch<any>, validate: boolean, touchOriginalTree: boolean): T;
+  export function applyPatch<T>(tree: T, patch: Patch<any>, validate = false, touchOriginalTree = true): T {
```

Comment:
@felixfbecker doesn't this signature alone suffice?
```ts
export function applyPatch<T>(tree: T, patch: Patch<any>, validate: boolean = false, touchOriginalTree: boolean = true): T {
  if (typeof validate !== 'boolean') { // if validate is not a boolean, it's being called from `reduce`
    validate = false;
    touchOriginalTree = true;
  }
```
According to [TS docs](https://www.typescriptlang.org/docs/handbook/functions.html), any param with a default value is optional.

> Default-initialized parameters that come after all required parameters are treated as optional, and just like optional parameters, can be omitted when calling their respective function.